lib/send-batch.ts

```diff
@@ -0,0 +1,81 @@
+import { chunk } from 'lodash'
+import { type Observable, catchError, concatMap, delay, from, map, of, retry, zip } from 'rxjs'
+import { type UserRow, useAppDataStore, useSendBatchState, useTemplateStore } from '~/lib/hooks'
+import { requestGoogleAccessToken } from '~/lib/token'
+import { type SendEmailParams, sendEmailAsync } from './email'
+
+const quotaLimit = 2
+const sendInterval = 1000
+
+type SendEmailStatus = 'success' | 'error'
+
+// Retry send email observable, this never emits error
+const mkSendEmailObservable = (params: SendEmailParams): Observable<SendEmailStatus> =>
+  from(sendEmailAsync(params)).pipe(
+    retry({
+      count: 2,
+      delay: 1000,
+    }),
+    map(() => 'success' as const),
+    catchError(() => of('error' as const))
+  )
+
+interface MkSlowSendBatchObservableInput {
+  users: UserRow[]
+  subjectTemplate: string
+  mdTemplate: string
+}
+
+export interface SendBatchProgress {
+  succeed: number
+  failed: number
+}
+
+export const mkSendBatchObservable = ({
+  users,
+  subjectTemplate,
+  mdTemplate,
+}: MkSlowSendBatchObservableInput): Observable<SendBatchProgress> => {
+  let succeed = 0
+  let failed = 0
+  return from(chunk(users, quotaLimit)).pipe(
```

Comment:
Instead of using `chunk`, I would send one message every 500 ms.

Reply:
> Instead of using `chunk`, I would send one message every 500 ms.

That would be easier do send every message 500 ms.